src/SignhostAPIClient/Rest/ErrorHandling/ErrorHandling.cs

```diff
@@ -8,31 +8,43 @@ public class ErrorHandling
 	{
 		public static void HandleError(HttpCall call)
 		{
-			string responseJson = call.Response.Content.ReadAsStringAsync().Result;
-			var error = JsonConvert.DeserializeAnonymousType(responseJson, new { Message = string.Empty });
+			string errorMessage = GetErrorMessage(call);
 
 			switch (call.HttpStatus) {
 				case HttpStatusCode.Unauthorized:
 					call.Exception = new System.UnauthorizedAccessException(
-						error.Message, call.Exception);
+						errorMessage, call.Exception);
 					break;
 				case HttpStatusCode.BadRequest:
 					call.Exception = new BadRequestException(
-						error.Message, call.Exception);
+						errorMessage, call.Exception);
 					break;
 				case HttpStatusCode.NotFound:
 					call.Exception = new NotFoundException(
-						error.Message, call.Exception);
+						errorMessage, call.Exception);
 					break;
 				case HttpStatusCode.InternalServerError:
 					call.Exception = new InternalServerErrorException(
-						error.Message, call.Exception, call.Response.Headers.RetryAfter);
+						errorMessage, call.Exception, call.Response.Headers.RetryAfter);
 					break;
 				default:
 					call.Exception = new SignhostRestApiClientException(
-						error.Message, call.Exception);
+						errorMessage, call.Exception);
 					break;
 			}
 		}
+
+		private static string GetErrorMessage(HttpCall call)
+		{
+			if (call?.Response?.Content?.ReadAsStringAsync()?.Result != null)
+			{
+				string responseJson = call.Response.Content.ReadAsStringAsync().Result;
```

Comment:
This can be optimized by moving string reponseJson assignment out of the if != null check. 
eg: string responseJson = call?.Response?.Content?.ReadAsStringAsync()?.Result;
if (responseJson != null)